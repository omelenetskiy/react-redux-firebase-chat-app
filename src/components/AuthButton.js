import React from 'react';
import PropTypes from 'prop-types';

const AuthButton = ({
  className, icon, buttonName, inProcess, onClick, title,
}) => (
    <button
      type="button"
      className={`ui-button ${className}`}
      onClick={(e) => {
        e.preventDefault();
        onClick(buttonName);
      }}
      title={title}
    >
      {inProcess ? <i className="fas fa-spinner fa-pulse" /> : buttonName || <i className={icon} />}
    </button>
  );

export default AuthButton;

AuthButton.defaultProps = {
  buttonName: '',
  inProcess: false,
  title: '',
  icon: null,
};

AuthButton.propTypes = {
  className: PropTypes.string.isRequired,
  buttonName: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  inProcess: PropTypes.bool,
  icon: PropTypes.string,
  title: PropTypes.string,
};
