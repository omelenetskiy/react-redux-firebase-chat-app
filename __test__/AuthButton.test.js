import React from 'react'
import { shallow } from 'enzyme'
import AuthButton from '../src/components/AuthButton'

describe('auth button', () => {
    let wrapper;
    let props;
    beforeEach(() => {
        props = { className: 'class', onClick: jest.fn() }
        wrapper = shallow(<AuthButton {...props} />);
    })
    it('render button', () => {
        console.log(wrapper.debug());
        expect(wrapper).toMatchSnapshot();
    });

    it('simulates click events', () => {
        const event = { preventDefault: jest.fn() }
        wrapper.find('button').simulate('click', event);
        expect(event.preventDefault).toHaveBeenCalledTimes(1);
        expect(props.onClick).toHaveBeenCalledTimes(1);
    });


})
