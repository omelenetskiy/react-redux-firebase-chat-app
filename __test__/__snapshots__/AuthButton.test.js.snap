// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth button render button 1`] = `
<button
  className="ui-button class"
  onClick={[Function]}
  title=""
  type="button"
>
  <i
    className={null}
  />
</button>
`;
